core/src/value/primitive.rs

```diff
@@ -604,6 +604,31 @@ impl PrimitiveValue {
         }
     }
 
+    /// Getter method to provide a clean version of a string or multiple strings,
+    /// removing unwanted whitespaces
+    pub fn to_clean_str(&self) -> Cow<str> {
+        match self {
+            PrimitiveValue::Str(values) => {
+                let new_value = values.trim_matches(|c| c == ' ' || c == '\u{0}');
+                Cow::from(new_value)
```

Comment:
We will need to use [`trim_end_matches`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.trim_end_matches) instead, because in some DICOM value representations, [leading spaces are assumed to be significant](http://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html).

```suggestion
                Cow::from(values.trim_end_matches(|c| c == ' ' || c == '\u{0}'))
```